AsyncDisplayKit/Details/ASPhotosFrameworkImageRequest.m

```diff
@@ -0,0 +1,169 @@
+//
+//  ASPhotosFrameworkImageRequest.m
+//  AsyncDisplayKit
+//
+//  Created by Adlai Holler on 9/25/15.
+//  Copyright © 2015 Facebook. All rights reserved.
+//
+
+#import "ASPhotosFrameworkImageRequest.h"
+#import "ASBaseDefines.h"
+
+NSString *const ASPhotosURLScheme = @"ph";
+
+static NSString *const _ASPhotosURLQueryKeyWidth = @"width";
+static NSString *const _ASPhotosURLQueryKeyHeight = @"height";
+
+// value is PHImageContentMode value
+static NSString *const _ASPhotosURLQueryKeyContentMode = @"contentmode";
+
+// value is PHImageRequestOptionsResizeMode value
+static NSString *const _ASPhotosURLQueryKeyResizeMode = @"resizemode";
+
+// value is PHImageRequestOptionsDeliveryMode value
+static NSString *const _ASPhotosURLQueryKeyDeliveryMode = @"deliverymode";
+
+// value is PHImageRequestOptionsVersion value
+static NSString *const _ASPhotosURLQueryKeyVersion = @"version";
+
+// value is 0 or 1
+static NSString *const _ASPhotosURLQueryKeyAllowNetworkAccess = @"network";
+
+static NSString *const _ASPhotosURLQueryKeyCropOriginX = @"crop_x";
+static NSString *const _ASPhotosURLQueryKeyCropOriginY = @"crop_y";
+static NSString *const _ASPhotosURLQueryKeyCropWidth = @"crop_w";
+static NSString *const _ASPhotosURLQueryKeyCropHeight = @"crop_h";
+
+@implementation ASPhotosFrameworkImageRequest
```

Comment:
What about ASPhotosFrameworkURLComponents, following the pattern from NSURLComponents? They serve essentially the same purpose and even have matching URL-conversion API

I hesitate to use Asset for a couple reasons, but they're not deal breakers:
- Photos.framework uses the word Asset to refer to metadata and Image to refer to data. PHAsset is a metadata struct and you use PHImageManager.requestImageForAsset to get the data
- Asset makes no distinction between video and image, but in its current form the class only supports requesting images and it could probably never be cleanly expanded to support video – we'd need a new class